games/speedy_snake.js

```diff
@@ -0,0 +1,306 @@
+..333333333333..
+.33333333333333.
+3333333333333333
+3333333333333333
+3333333333333333
+3333333333333333
+3333333333333333
+3333333333333333
+3333333333333333
+.33333333333333.
+..333333333333..
+...3333333333...
+....33333333....`],
+);
+
+setSolids([]);
+
+let level = 0;
+const levels = [
+  map`
+wwwwwwwwwwwwwwwwwww  
+wwwwwwwwwwwwwwwwwww  
+wwwwwwwwwwwwwwwwwww  
+wwwwwwwwwwwwwwwwwww  
+wwwwwwwwwwwwwwwwwww  
+wwwwwwwwwwwwwwwwwww 
+w.................w 
+w.................w
+w...p.............w
+w.................w
+w.................w
+w.................w
+w.................w
+w.................w
+w.................w
+w.................w
+w.................w
+w.................w
+w.................w
+w.................w
+w.......a.........w
+w.................w
+w.................w
+wwwwwwwwwwwwwwwwwww
+`,
+  //17x17 grid of open space, not including walls
+];
+
+let score=0;
+addText("score: "+score, 
+  {x:4, 
+   y:2, 
+   color: color`2` })
+addText("Speedy Snake", 
+  {x:4, 
+   y:1, 
+   color: color`4` })
+
+setMap(levels[level]);
+
+var g=null;
+var dir="d";
+var speed=150;
+var x=3;
+var mode="easy";
+var pausePlay=1;
+var gameO=0;
+
+onInput("s", () => {
+  dir="d";
+});
+
+onInput("w", () => {
+  dir="u";
+});
+
+onInput("a", () => {
+  dir="l";
+});
+
+onInput("d", () => {
+  dir="r";
+});
+onInput("l", () => {
+  resetGame();
+});
+onInput("j", () => {
+  mode="hard";
+  resetGame();
+});
+onInput("k", () => {
+  mode="easy";
+  resetGame();
+}); 
+onInput("i", () =>{
+  pause();
+});
+
+const appleTune = tune`
+283.0188679245283,
+283.0188679245283: f5~283.0188679245283,
+8490.566037735849`
+const wallTune = tune `
+145.63106796116506,
+145.63106796116506: c5~145.63106796116506,
+145.63106796116506: c5~145.63106796116506,
+145.63106796116506: b4~145.63106796116506,
+145.63106796116506: b4~145.63106796116506,
+145.63106796116506: a4~145.63106796116506,
+145.63106796116506: a4~145.63106796116506,
+3640.7766990291266`
+
+function resetApple(){
+  playTune(appleTune);
+  getFirst(apple).x=Math.floor(Math.random()*16)+1;
+  getFirst(apple).y=Math.floor(Math.random()*15)+6;
+  score++;
+  addText("score: "+score, 
+          {x:4, 
+           y:2, 
+           color: color`2` })
+  if(mode=="easy"){
+    x=3;
+  }
+  else if(mode=="hard"){
+    x=5;
+  }
+  speed-=x;
+  setI();
+}
+
+function gameOver(){
+  clearInterval(g);
+  gameO=1;
+  playTune(wallTune);
+  if(dir=="d"){
+    getFirst(player).y--;
+  } else if(dir=="u"){
+    getFirst(player).y++;
+  } else if(dir=="r"){
+    getFirst(player).x--;
+  } else if(dir=="l"){
+    getFirst(player).x++;
+  }
+  addText("Game Over!", 
+  {x:5, 
+   y:10, 
+   color: color`3` })
+  addText("L=restart", 
+  {x:6, 
+   y:12, 
+   color: color`3` })
+}
+
+function checkState(){
+  var arr = getTile(getFirst(player).x,getFirst(player).y)
+  if(arr.length>1){
+    if(arr[1].type=="a"){
+      resetApple();
+    }
+  } else{
+    //console.log(getFirst(player).y, getFirst(player).x)
```

Comment:
what is the functionality of this commented code?

Reply:
The console.log was just for debugging, I can remove it now that it works fine!